CTOR-1198 : Refresh applications-biztalk (#3852)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/applications-biztalk.md

id: applications-biztalk
title: MS Biztalk
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Connector dependencies

The following monitoring connectors will be installed when you install the **MS Biztalk** connector through the
**Configuration > Monitoring Connector Manager** menu:
* [Base Pack](./base-generic.md)

## Pack assets

### Templates

The Monitoring Connector **MS Biztalk** brings a host template:

* **App-Biztalk-custom**

The connector brings the following service templates (sorted by the host template they are attached to):

<Tabs groupId="sync">
<TabItem value="App-Biztalk-custom" label="App-Biztalk-custom">

| Service Alias               | Service Template                               | Service Description                                       |
|:----------------------------|:-----------------------------------------------|:----------------------------------------------------------|

> | Received-Locations-Disabled | App-Biztalk-Received-Locations-Disabled-custom | Check the number of "biztalk received locations" disabled |

Reviewer: ```suggestion
| Received-Locations-Disabled | App-Biztalk-Received-Locations-Disabled-custom | Check the number of disabled 
 "biztalk received locations" |
```

Reviewer: receive locations?

> The services listed above are created automatically when the **App-Biztalk-custom** host template is used.

</TabItem>
</Tabs>

### Collected metrics & status

Here is the list of services for this connector, detailing all metrics and statuses linked to each service.

<Tabs groupId="sync">
<TabItem value="Received-Locations-Disabled" label="Received-Locations-Disabled">

| Name  | Unit  |
|:------|:------|
| count | count |

</TabItem>
</Tabs>

## Prerequisites

> Information about the **receive locations** are stored in the `adm_ReceiveLocation` table within 

Reviewer: ```suggestion
* Information about the **receive locations** are stored in the `adm_ReceiveLocation` table within 
```

> the BizTalk Management Database (BizTalkMgmtDb). Therefore, you need a SQL account with the necessary 

Reviewer: ```suggestion
the BizTalk Management Database (BizTalkMgmtDb). Therefore, you need an SQL account with the necessary 
```

> permissions, at least the db_datareader role on the BizTalkMgmtDb database. You must also ensure that 

Reviewer: ```suggestion
permissions, at least the db_datareader role on the BizTalkMgmtDb database.
* You must also ensure that 
```

> the Centreon poller has access to the SQL server hosting the BizTalkMgmtDb database. Verify the 

Reviewer: ```suggestion
the Centreon poller has access to the SQL server hosting the BizTalkMgmtDb database.
* Check that the 
```

| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options). | --verbose         |             |

</TabItem>
</Tabs>

3. [Deploy the configuration](/docs/monitoring/monitoring-servers/deploying-a-configuration). The service appears in the list of services, and on the **Resources Status** page. The command that is sent by the connector is displayed in the details panel of the service: it shows the values of the macros.

## How to check in the CLI that the configuration is OK and what are the main options for?

Once the plugin is installed, log into your Centreon poller's CLI using the
**centreon-engine** user account (`su - centreon-engine`). Test that the connector 
is able to monitor a resource using a command like this one (replace the sample values by yours):

```bash
/usr/lib/centreon/plugins/centreon_biztalk.pl \
	--plugin=database::mssql::plugin \
	--dyn-mode=apps::biztalk::sql::mode::rlocationdisabled \
	--hostname=10.0.0.1 \
	--port=1433 \
	--username='' \

> 	--password='' \

Reviewer: username and password

Reviewer 2: Ne sont pas obligatoires pour lancer le plugin : 
```
perl centreon_plugins.pl --plugin=database::mssql::plugin --dyn-mode=apps::biztalk::sql::mode::rlocationdisabled --hostname=10.0.0.1
UNKNOWN: Cannot connect: OpenClient message: LAYER = (0) ORIGIN = (0) SEVERITY = (78) NUMBER = (44)
Server , database 
Message String: Server name not found in configuration files.
OpenClient message: LAYER = (0) ORIGIN = (0) SEVERITY = (78) NUMBER = (45)
Server , database 
Message String: Unknown host machine name.
OpenClient message: LAYER = (0) ORIGIN = (0) SEVERITY = (78) NUMBER = (41)
Server , database 
Message String: Unable to connect: Adaptive Server is unavailable or does not exist 
```
